### PR TITLE
Adding developer documentation outlining recommended application stru…

### PR DESCRIPTION
**Pull request by tpmccallum**
…cture

Content must go through a pre-merge checklist.

## Pre-Merge Content Checklist

This documentation has been checked to ensure that:

- [x] The `title, `template`, and `date` are all set
- [x] Does this PR have a new menu item (anywhere in `templates/*.hbs` files) that points to a document `.md` that is set to publish in the future? If so please only publish the `.md` and `.hbs` changes in real-time (otherwise there will be a menu item pointing to a `.md` file that does not exist)
- [x] File does not use CRLF, but uses plain LF (hint: use `cat -ve <filename> | grep '^M' | wc -l` and expect 0 as a result) 
- [x] Has passed [`bart check`](https://developer.fermyon.com/bartholomew/quickstart)
- [x] Has been manually tested by running in Spin/Bartholomew (hint: use `PREVIEW_MODE=1` and run `npm run styles` to update styling)
- [x] Headings are using Title Case
- [x] Code blocks have the programming language set to properly highlight syntax and the proper copy directive
- [x] Have tested with `npm run test` and resolved all errors
- [x] Relates to an existing (potentially outdated) blog article? If so please add URL in blog to point to this content.

### CODE REVIEW

File: content/spin/spin-application-structure.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+template = "spin_main"
+date = "2023-08-20T00:00:00Z"
+enable_shortcodes = true
+[extra]
+url = "https://github.com/fermyon/developer/blob/main/content/spin/spin-application-structure.md"
+keywords = "structure"
+
+---
+
+As mentioned in this [blog post](https://www.fermyon.com/blog/spin-application-structure), we are often asked about how to structure Spin applications. There are a few nuances to this discussion, such as the use of different application trigger types and how to modify an existing application's structure. These nuances are discussed in the blog post and also in [this video](https://www.youtube.com/watch?v=QQD-qodabSc). This page, however, will focus only on creating Spin applications from scratch using the http-empty template. This template serves as the foundation for the recommended Spin application structure when creating applications that respond to HTTP requests.
```
Comment (itowlson) [inline content/spin/spin-application-structure.md:11]: This extra structure scaffolded by `http-empty` is mainly aimed at apps with multiple custom components so maybe just allude to that?

```suggestion
As mentioned in this [blog post](https://www.fermyon.com/blog/spin-application-structure), we are often asked about how to structure Spin applications that involve multiple source code components. There are a few nuances to this discussion, such as the use of different application trigger types and how to modify an existing application's structure. These nuances are discussed in the blog post and also in [this video](https://www.youtube.com/watch?v=QQD-qodabSc). This page, however, will focus only on creating multi-component Spin applications from scratch using the http-empty template. This template serves as the foundation for the recommended Spin application structure when creating applications that respond to HTTP requests.
```

File: templates/spin_sidebar.hbs [hunk 2]
```diff
@@ -143,6 +143,9 @@
                     Advanced
                 </label>
                 <ul class="menu-list accordion-menu-item-content">
+                    <li><a {{#if (active_content request.spin-path-info "/spin/spin-application-structure" )}} class="active"
```
Comment (itowlson) [inline templates/spin_sidebar.hbs:146]: Consider putting this under Applications rather than Advanced?

**Review by itowlson — APPROVED** [review]
LGTM, just a couple of non-blocking suggestions

File: content/spin/spin-application-structure.md (hunk 1 above)
Comment (tpmccallum) [reply content/spin/spin-application-structure.md:11]: Thank you @itowlson; great point!
Changes in new push to branch.

File: templates/spin_sidebar.hbs [hunk 3]
```diff
@@ -143,6 +143,9 @@
...
+                            {{/if}} href="{{site.info.base_url}}/spin/spin-application-structure">Spin Application Structure</a>
```
Comment (itowlson) [inline templates/spin_sidebar.hbs:147]: Sorry for missing this earlier but it occurred to me we can drop "Spin" given this is part of the Spin docs (same in the page title)
```suggestion
                            {{/if}} href="{{site.info.base_url}}/spin/spin-application-structure">Application Structure</a>
```

Comment (tpmccallum) [reply templates/spin_sidebar.hbs]: Thanks; I have updated the menu and the page title to be more uniform with what already exists.

**Pull request merged**